### PR TITLE
Feature/report benchmarks

### LINKED ISSUE #2427: Create aggregate report benchmark suite
The (aggregate) reports are slow when we put in a lot of objects. This has several aspects as discussed in #2446, and the goal is to improve the performance significantly while keeping the structure clear. This also means keeping track properly of the performance of the reports over time with profiling. Hence, we need to introduce a way to benchmark report creation structurally so we can verify these improvements.

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
This adds a make target in rocky that calls a slow integration tests and saves a pstat file of the test. This test is ignored by default.

### Issue link

Closes #2427 

### Demo

Scherm­afbeelding 2024-02-06 om 15 01 50

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -442,3 +442,5 @@ nl-kat-*
 # rpki cache
 /boefjes/boefjes/plugins/kat_rpki/rpki.json
 /boefjes/boefjes/plugins/kat_rpki/rpki-meta.json
+
+bench*.pstat
```
Comment (ammar92) [inline .gitignore:446]: I think in general we can skip profilers `.pstat` files
```suggestion
*.pstat
```

File: rocky/Makefile [hunk 2]
```diff
@@ -34,17 +34,20 @@ run:
...
+
+utest: testclean ## Run the unit tests.
+	docker compose  -f .ci/docker-compose.yml run --rm rocky_tests
+
+itest: testclean ## Run the integration tests.
+	docker compose  -f .ci/docker-compose.yml run --rm rocky_integration
+
+bench: testclean ## Run the report benchmark.
+	docker compose  -f .ci/docker-compose.yml run --rm rocky_integration \
+	python -m cProfile -o .ci/bench_$$(date +%Y_%m_%d-%H:%M:%S).pstat -m pytest -m slow --no-cov tests/integration
```
Comment (ammar92) [inline rocky/Makefile:50]: Nice one, a good example on how to use task dependencies. I would only suggest `clean` as an alternative task name, but I'll leave it up to you.

File: rocky/tests/integration/conftest.py [hunk 3]
```diff
@@ -47,22 +47,27 @@ def octopoes_api_connector_2(request) -> OctopoesAPIConnector:
...
 
     hostnames = [
-        Hostname(network=network.reference, name="example.com"),
-        Hostname(network=network.reference, name="a.example.com"),
-        Hostname(network=network.reference, name="b.example.com"),
-        Hostname(network=network.reference, name="c.example.com"),
-        Hostname(network=network.reference, name="d.example.com"),
-        Hostname(network=network.reference, name="e.example.com"),
-        Hostname(network=network.reference, name="f.example.com"),
+        Hostname(network=network.reference, name=f"{test_hostname}"),
```
Comment (ammar92) [inline rocky/tests/integration/conftest.py:60]:
```suggestion
        Hostname(network=network.reference, name=test_hostname),
```

File: rocky/tests/integration/conftest.py [hunk 4]
```diff
@@ -47,22 +47,27 @@ def octopoes_api_connector_2(request) -> OctopoesAPIConnector:
     connector.delete_node()
 
 
-def seed_system(octopoes_api_connector: OctopoesAPIConnector, valid_time):
+def seed_system(
+    octopoes_api_connector: OctopoesAPIConnector,
+    valid_time,
```
Comment (ammar92) [inline rocky/tests/integration/conftest.py:52]:
```suggestion
    valid_time: datetime.datetime,
```

**Review by ammar92 — APPROVED** [review]
Small suggestions, looks good otherwise

File: rocky/Makefile (hunk 2 above)
Comment (Donnype) [reply rocky/Makefile:50]: Could be a bit confusing shadowing the main `make clean` we have in the base directory I think

**Review by ammar92 — APPROVED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
All seem to work as expected. Could you type a few more lines in the ticket though? Let's try to prevent empty tickets as it's good to have a trail of what/why we are doing things. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by dekkers — APPROVED** [review]

**Pull request merged**